sagemaker-clarify/fairness_and_explainability/fairness_and_explainability_aws_sdk.ipynb

```diff
@@ -0,0 +1,889 @@
+    "  * InputName dataset which we are providing here as an Amazon S3 object.\n",
+    "* ProcessingOutputConfig: The job also requires an output parameter, the output location as an Amazon S3 prefix with the OutputName \"analysis_result\". The S3UploadMode should be set to \"EndOfJob\", because the analysis results is generated at the end of the job. We will be providing here the `analysis_result_path` that we configured earlier.\n",
+    "* ProcessingResources contains the ClusterConfig specifying the ML compute instance_type we want to use and the count. SageMaker SHAP analysis is CPU-intensive, using a better instance type should speed up the analysis. The SageMaker Clarify job doesn’t use GPU.\n",
+    "* StoppingCondition: Using a maximum limit of 30 min for example job run. You can set the MaxRuntimeInSeconds of a SageMaker Clarify job to up to 7 days (604800 seconds). If the job cannot be completed within this time limit, it will be force-stopped and no analysis results are provided."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "processing_job_name = create_processing_job(bias_analysis_config_path)\n",
+    "\n",
+    "# Wait for processing job to complete\n",
+    "status = sagemaker_client.describe_processing_job(ProcessingJobName=processing_job_name)\n",
+    "\n",
+    "while status[\"ProcessingJobStatus\"] == \"InProgress\":\n",
+    "    status = sagemaker_client.describe_processing_job(ProcessingJobName=processing_job_name)\n",
+    "    print(status[\"ProcessingJobStatus\"])\n",
+    "    time.sleep(60)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Viewing the Bias Report\n",
+    "In Studio, you can view the results under the experiments tab.\n",
+    "\n",
+    "<img src=\"./recordings/bias_report.gif\">\n",
+    "\n",
+    "Each bias metric has detailed explanations with examples that you can explore.\n",
+    "\n",
+    "<img src=\"./recordings/bias_detail.gif\">\n",
+    "\n",
+    "You could also summarize the results in a handy table!\n",
+    "\n",
+    "<img src=\"./recordings/bias_report_chart.gif\">\n",
+    "\n",
+    "If you're not a Studio user yet you can access the complete analysis report at the following S3 bucket.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "analysis_result_path"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Explaining Predictions\n",
+    "There are expanding business needs and legislative regulations that require explanations of _why_ a model made the decision it did. SageMaker Clarify uses SHAP to explain the contribution that each input feature makes to the final decision."
```

Comment:
SHAP => Kernel SHAP.

Reason: to be more specific. SHAP a unified framework for interpreting the predictions of any machine learning model, it has extensions like Kernel SHAP and Tree SHAP. Currently SageMaker Clarify only supports Kernel SHAP.